### ISSUE TITLE
Provide a tool to inject the conan cmake "magic" lines

### ISSUE THREAD

**Issue by lasote** [conan-io/conan#2590]
Feature request:

To simplify the recipes with the classical `replace_in_file`, we could provide a helper, could be inside the `CMake` helper as a static method (?) that searches the `project()` in a `CMakeLists.txt`and inject the lines:
```
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
CONAN_BASIC_SETUP()
```

Proposed: `CMake.insert_conan_build_info(folder)` 
Maybe it could detect the `install_folder` to adjust the install dir if present?

**Comment by jgsogo** [comment]
Packaging apache-apr I found that I needed to put those magic lines down in the `CMakeLists.txt` in order to work, [here](https://github.com/jgsogo/conan-apache-apr/blob/7d0435c27e7a417238fb8d25af33d31bd37ab426/conanfile.py#L31) is the `replace_in_file`. Maybe I'm doing something wrong or maybe it won't fit every recipe... 

If I were to propose something _standard_ in that direction I will take a look at the @bincrafters approach with the `cmake_wrapper` project in a different `CMakeLists.txt`, it is less intrusive and it works ([e.g. here](https://github.com/bincrafters/conan-openal/blob/stable/1.18.2/CMakeLists.txt)). I haven't tried that approach in the apache-apr recipe, it may not work, so it won't be _standard_, but I like the idea of not modifying a file (although I don't like having an additional file). :loop:
